source/device/image_client.py

```diff
@@ -100,8 +107,8 @@ def read_next_frames(self):
 		ticket, answer = self.read_next_answer()
 
 		if ticket == b"0000":
-			delimiter = str(answer).find('stop')
-			result = self._deserialize_image_chunk(data=answer[delimiter-8:])
+			delimiter = answer.find(b'stop')
+			result = self._deserialize_image_chunk(data=answer[delimiter+4:])
 			self.frames = [result[i][1] for i in result]
```

Comment:
I feel that it would be nice to have checks for find as to not have `IndexError`

Reply:
Done... admittedly I didn't choose a very elaborate solution (I would leave that to @Galoshi later on) but the error message clearly points you to the issue and the `IndexError` is avoided.